File: project/jimmer-sql-test-model/src/main/java/org/babyfish/jimmer/sql/model/microservice/Product.java
package org.babyfish.jimmer.sql.model.microservice;

import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.Id;
import org.babyfish.jimmer.sql.ManyToMany;
import org.babyfish.jimmer.sql.Table;

import java.util.List;

@Entity(microServiceName = "product-service")
@Table(name = "MS_PRODUCT")
public interface Product {

    @Id
    long id();

    String name();

    @ManyToMany(mappedBy = "products")
    List<OrderItem> orderItems();
}

File: project/jimmer-sql-kotlin-test-model/src/main/kotlin/org/babyfish/jimmer/sql/kt/model/microservice/Product.kt
package org.babyfish.jimmer.sql.kt.model.microservice

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.ManyToMany
import org.babyfish.jimmer.sql.Table

@Entity(microServiceName = "product-service")
@Table(name = "MS_PRODUCT")
interface Product {
    
    @Id
    val id: Long
    
    val name: String

    @ManyToMany(mappedBy = "products")
    val orderItems: List<OrderItem>
}